helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -748,6 +766,12 @@ public void handleChildChange(String parentPath, List<String> currentChilds) {
           // removeListener will call handler.reset(), which in turn call invoke() on FINALIZE type
           _manager.removeListener(_propertyKey, _listener);
         } else {
+          if (!isReady()) {
+            // avoid leaking CallbackHandler
+            logger.info("Callbackhandler {} with path {} end of life as not ready to avoid leak",
```

Comment:
nit, could you please rephrase this? I think we need to mention the re-subscribe is skipped because of the CH end of life.

Reply:
good point. changed to "Callbackhandler {} with path {} is in reset state. Stop subscription to ZK client to avoid leaking"